mlos_bench/mlos_bench/storage/base_trial_data.py

```diff
@@ -46,38 +48,53 @@ def exp_id(self) -> str:
     @property
     def trial_id(self) -> int:
         """
-        ID of the current trial.
+        ID of the trial.
         """
         return self._trial_id
 
     @property
     def config_id(self) -> int:
         """
-        ID of the configuration of the current trial.
+        ID of the configuration of the trial.
         """
         return self._config_id
 
     @property
     def ts_start(self) -> datetime:
         """
-        Start timestamp of the current trial (UTC).
+        Start timestamp of the trial (UTC).
         """
         return self._ts_start
 
     @property
     def ts_end(self) -> Optional[datetime]:
         """
-        End timestamp of the current trial (UTC).
+        End timestamp of the trial (UTC).
         """
         return self._ts_end
 
     @property
     def status(self) -> Status:
         """
-        Status of the current trial.
+        Status of the trial.
         """
         return self._status
 
+    def _df_to_dict(self, dataframe: pandas.DataFrame) -> Dict[str, Optional[TunableValue]]:
+        """Utility function to convert certain key-value dataframe formats to a dict."""
+        if dataframe.columns.tolist() == ['metric', 'value']:
+            dataframe = dataframe.copy()
+            dataframe.rename(columns={'metric': 'parameter'}, inplace=True)
+        assert dataframe.columns.tolist() == ['parameter', 'value']
+        data = {}
+        for _, row in dataframe.iterrows():
+            assert isinstance(row['parameter'], str)
+            assert row['value'] is None or isinstance(row['value'], (str, int, float))
```

Comment:
```python
from types import NoneType
```
and
```suggestion
            assert isinstance(row['value'], (NoneType, str, int, float))
```

Reply:
We've had issues with NoneType support across python versions in the past, so I skipped this for now.